docs/03_goclient/15_sessions.md

```diff
@@ -0,0 +1,134 @@
+  HostName          string
+
+  // ... other unexported fields
+}
+
+func GetSessionInfo(ctx Context) *SessionInfo
+```
+
+The session context also stores some metadata of the session, which can be retrieved by the `GetSessionInfo()` API. For now, there are only two exported fields in the returned `workflow.SessionInfo`. We may consider exposing more information in the future.
+
+## Concurrent Session Limitation
+
+It's very easy to limit the number of concurrent sessions running on a worker, just set the `MaxConcurrentSessionExecutionSize` field of `worker.Options` to the desired value. By default this field is set to a very large value, so there's no need to manually set it if no limitation is needed.
+
+If a worker hits this limitation, it won't accept any new `CreateSession()` request until one of the existing sessions is completed and `CreateSession()` will return an error if the session can't be created within `CreationTimeout`.
+
+## Recreate Session
+
+For long-running sessions, user may want to use the ContinueAsNew feature to split the workflow into multiple runs, but still needs all the activities to be executed by the same worker. The `RecreateSession()`  API is designed for such use case.
+
+```go
+func RecreateSession(ctx Context, recreateToken []byte, sessionOptions *SessionOptions) (Context, error)
+```
+
+Its usage is the same as `CreateSession()` except that it also takes in a `recrateToken`, which is needed to create a new session on the same worker as the previous one. User can get the token by calling the `GetRecreateToken()` method of the `SessionInfo` object and pass it to the next run.
+
+```go
+token := workflow.GetSessionInfo(sessionCtx).GetRecrateToken()
+```
+
+## Q&A
+
+### Is there a complete sample code?
+Yes, the fileprocessing example in the cadence-sample repo has been updated to use the session framework.
+
+### What happens to my activity if worker dies?
+If your activity has already been scheduled, it will be cancelled. If not, you will get an `workflow.ErrSessionFailed` error when you call `workflow.ExecuteActivity()`.
+
+### Is the concurrent session limitation per process or per host?
+It's per worker process, so make sure there's only one worker process running on the host if you plan to use that feature.
+
+
+## Future Work
+
+* **Support automatic session re-establishing**:
+  Right now a session is considered failed if the worker process dies. However, for some use cases, user only cares if the worker host is alive or not. For these uses cases, we should automatically re-establish the session if the worker process is restarted.
+
+* **Support fine-grained concurrent session limitation**:
+  The current implementation assumes that all sessions are consuming the same type of resource and there's only one global limitation. In the future, we may allow users to specify what type of resource their session will consume and enforce different limitations on different type of resources.
```

Comment:
"In the future, we may allow users" -> "Our plan is to allow you"

"their session" -> "your session"

"different type" -> "different types"